ONSdigital/dp-cantabular-dimension-api
# Feature/get area types

### What

- Implement `GET /area-types` endpoint, which returns a list of geography dimensions from cantabular (using graphQL)
- Add config and healthcheck checker for api ext
- Fixed component tests and add tests for the new endpoint
- Removed `/hello` endpoint
___
- Depends on this pr being merged first: https://github.com/ONSdigital/dp-cantabular-dimension-api/pull/2
- Depends on this dp-api-clients-go pr: https://github.com/ONSdigital/dp-api-clients-go/pull/178
- [Trello card](https://trello.com/c/tSdkHCF2/5452-get-area-types-implement-endpoint-8)

### How to review

- Make sure code changes make sense
- Make sure unit/component tests pass

(tested with dp-compose - already tested, but feel free to verify)
- run the latest dp-compose cantabular and try:
  - `GET http://localhost:27200/area-types?cantabular_blob=Example` (with florence token) -> 200
  - `GET http://localhost:27200/area-types` (with florence token) -> 200
  - `GET http://localhost:27200/area-types` (without florence token) -> 401
  - `GET http://localhost:27200/area-types?cantabular_blob=wrong` (with florence token) -> 404
  - `GET http://localhost:27200/health` (without florence token) -> 200

### Who can review

Anyone
